diff --git a/common/drawable_desc.c b/common/drawable_desc.c
index a7e4237..9b2c3ab 100644
--- a/common/drawable_desc.c
+++ b/common/drawable_desc.c
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <xorg-server.h>
 
 #include "pixmaputil.h"
 
diff --git a/common/get_first_pixel.c b/common/get_first_pixel.c
index 662ab65..f866893 100644
--- a/common/get_first_pixel.c
+++ b/common/get_first_pixel.c
@@ -1,3 +1,4 @@
+#include <xorg-server.h>
 #include "pixmapstr.h"
 #include "pixmaputil.h"
 
diff --git a/common/picture_desc.c b/common/picture_desc.c
index 8ffbc84..4b988a5 100644
--- a/common/picture_desc.c
+++ b/common/picture_desc.c
@@ -1,3 +1,4 @@
+#include <xorg-server.h>
 #include "picturestr.h"
 #include "pixmaputil.h"
 #include "pictureutil.h"
diff --git a/common/picture_solid.c b/common/picture_solid.c
index 88e23ab..a71fc90 100644
--- a/common/picture_solid.c
+++ b/common/picture_solid.c
@@ -1,3 +1,4 @@
+#include <xorg-server.h>
 #include "picturestr.h"
 #include "pixmapstr.h"
 #include "pictureutil.h"
diff --git a/common/transform.c b/common/transform.c
index 07c79be..e7c7108 100644
--- a/common/transform.c
+++ b/common/transform.c
@@ -1,3 +1,4 @@
+#include <xorg-server.h>
 #include "picturestr.h"
 #include "pictureutil.h"
 
